src/lib/snark_worker/entry.ml

```diff
@@ -167,25 +167,29 @@ let main (module Rpcs_versioned : Intf.Rpcs_versioned_S) ~logger ~proof_level
           ~metadata:[ ("time", `Float random_delay) ] ;
         let%bind () = wait ~sec:random_delay () in
         go ()
-    | Ok (Some (work, public_key)) -> (
+    | Ok (Some partitioned_spec) -> (
+        let address_json =
+          ("address", `String (Host_and_port.to_string daemon_address))
+        in
+        let work_ids_json =
+          ( "work_ids"
+          , `One (Spec.Partitioned.Stable.Latest.statement partitioned_spec)
```

Comment:
`One` looks a bit odd

Reply:
This is reusing the old code, I'll try to get rid of them